src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPDocumentManager.cs

```diff
@@ -92,10 +93,15 @@ public override void UntrackDocument(ITextBuffer buffer)
                 return;
             }
 
-            _documents.Remove(uri);
-
-            var args = new LSPDocumentChangeEventArgs(lspDocument.CurrentSnapshot, @new: null, LSPDocumentChangeKind.Removed);
-            Changed?.Invoke(this, args);
+            if (_documents.TryRemove(uri, out _))
+            {
+                var args = new LSPDocumentChangeEventArgs(lspDocument.CurrentSnapshot, @new: null, LSPDocumentChangeKind.Removed);
+                Changed?.Invoke(this, args);
+            }
+            else
+            {
+                Debug.Fail($"Couldn't remove {uri.AbsolutePath}. This should never ever happen.");
```

Comment:
Maybe we don't need it for this PR, but it seems like we've got a lot of spots in this class that should never be hit. I think it would be good if we added a logger to this class and wrote an error log if they DID get hit so we had more info.

Reply:
True. Unfortunately, there is not a good logging story in the VS Client like in VSCode. All we can do is the Activity logger which requires the main thread, slows things down and isn't very convenient. For now, we have to stick with `Debug.Fail`s and exceptions.